streaming/src/main/scala/org/apache/spark/streaming/Checkpoint.scala

```diff
@@ -255,19 +263,24 @@ object CheckpointReader extends Logging {
         // loader to find and load classes. This is a well know Java issue and has popped up
         // in other places (e.g., http://jira.codehaus.org/browse/GROOVY-1627)
         val zis = compressionCodec.compressedInputStream(fis)
-        val ois = new ObjectInputStreamWithLoader(zis,
-          Thread.currentThread().getContextClassLoader)
-        val cp = ois.readObject.asInstanceOf[Checkpoint]
-        ois.close()
-        fs.close()
-        cp.validate()
+        ois = new ObjectInputStreamWithLoader(zis,
+            Thread.currentThread().getContextClassLoader)
+        cpOption = Some(ois.readObject.asInstanceOf[Checkpoint])
         logInfo("Checkpoint successfully loaded from file " + file)
-        logInfo("Checkpoint was generated at time " + cp.checkpointTime)
-        return Some(cp)
       } catch {
         case e: Exception =>
           logWarning("Error reading checkpoint from file " + file, e)
+      } finally {
+        if (ois != null) {
+          ois.close()
+        }
+        fs.close()
+      }
+      cpOption.foreach { cp =>
+        cp.validate()
+        logInfo("Checkpoint was generated at time " + cp.checkpointTime)
       }
+      return cpOption
```

Comment:
I think this changes the semantics. before, an exception should ultimately mean a `SparkException. Now, you return `None`. In fact isn't the final statement unreachable now since the `finally` block returns?